impl/client/src/main/javascript/web/pentaho/type/action/base.js

```diff
@@ -764,23 +1004,74 @@ define([
         this.__result = undefined;
       },
 
-      // endregion
+      /**
+       * Rejects the action with a given reason and,
+       * if that fails due to a cancellation not currently being a valid operation,
+       * fails the action.
+       *
+       * @param {string|Error} [reason] - The reason for the rejection.
+       *
+       * @private
+       */
+      __rejectFinal: function(reason) {
+        try {
+          this.__reject(reason);
+        } catch(ex) {
+          // Reason was a cancellation, which is invalid during do. Only failure, allowed.
+          this.__reject(ex);
+        }
+      },
 
-      __doPhaseInit: function() {
+      // region Private __executePhase* Methods
+      /**
+       * Executes the **init** phase.
+       *
+       * Changes the state to [init]{@link pentaho.type.action.States.init}
+       * and delegates to [_onPhaseInit]{@link pentaho.type.action.Base#_onPhaseInit}.
+       *
+       * Used by both the synchronous and the asynchronous action kinds.
```

Comment:
"kinds" seems superfluous. What about: "Used by both synchronous and asynchronous actions."